### PR TITLE
Implement basic reward service

### PR DESCRIPTION
**Pull request by PaulBredl**
## Description of changes
Implement basic reward service, currently without automatic tests and only health is implemented.

  
## How has this been tested?
Please describe the test strategy you followed.

- [ ] automated integration test
- [ ] automated unit test
- [ ] automated acceptance test
- [x] manual, exploratory test

- Use the gits_backend
- Rebuild everything,  clear volumes -> db should be empty
- create a user in keycloak and store the user id somewhere
- run the script to create sample data
- go to graphiql of the reward service, check the health score for your created user and course from the sample data
- recalculate the scores -> should be now 80 because of the contents to do
- there is one content with only one media record -> find out its id and then the id of the media record
- use the mutation of the media service to log user progress for this media record -> query from the gateway using a valid auth header
- query the rewards scores again -> should now be increased to around 86.
    
## Checklist before requesting a review
- [x] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My code fulfills all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [x] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/gits/reward/service/RewardService.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        AllRewardScoresEntity allRewardScoresEntity = rewardScoresRepository
+                .findById(new AllRewardScoresEntity.PrimaryKey(courseId, userId))
+                .orElseGet(() -> initializeRewardScores(courseId, userId));
+
+        List<UUID> chapterIds = courseServiceClient.getChapterIds(courseId);
+
+        List<Content> contents = contentServiceClient.getContentsWithUserProgressData(userId, chapterIds);
+
+        allRewardScoresEntity
+                .setHealth(new HealthScoreCalculator().recalculateScore(allRewardScoresEntity.getHealth(), contents));
```
Comment (myluki2000) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/RewardService.java:49]: This is ugly. IMO, as the ScoreCalculator interface is never used, it's unnecessary. Why not just use static methods?

Or if you really want that interface, use singletons.

Comment (PaulBredl) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/RewardService.java:49]: I can add them as constructor fields if you like, static methods are pretty ugly too, e.g. if we want to mock that stuff or want to have different calculation methods in the future or whatever. 
The interface is unnecessary but also it kinda makes sense that all of them work the same so I thought an interface makes sense

Comment (myluki2000) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/RewardService.java:49]: And why not privatize the constructor and provide a getInstance() singleton method? Then we can still have the interface in case we want to pass around calculators in methods later on and we can also mock the stuff.

Comment (PaulBredl) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/RewardService.java]: Done as discussed in discord

**Review by myluki2000 — APPROVED** [review]
